### PR TITLE
Adding new status field and refactoring old field

### PR DESCRIPTION
**Pull request by chrislovecnm**
- refactored old status field to operatorStatus
- including new status feild for OpenShift

### CODE REVIEW

File: config/crd/bases/crdb.cockroachlabs.com_crdbclusters.yaml
Comment (chrislovecnm) [inline config/crd/bases/crdb.cockroachlabs.com_crdbclusters.yaml:355]: TODO - this should have changed to operatorConditions

File: pkg/actor/deploy.go
Comment (chrisseto) [inline pkg/actor/deploy.go:67]: left over println?

File: pkg/actor/deploy.go
Comment (chrisseto) [inline pkg/actor/deploy.go:64]: Use `ctx` here?

File: pkg/actor/deploy.go
Comment (chrisseto) [inline pkg/actor/deploy.go:164]: use `ctx`?

File: pkg/actor/deploy.go
Comment (chrisseto) [inline pkg/actor/deploy.go:167]: left over `println`?

File: pkg/actor/deploy_test.go
Comment (chrisseto) [inline pkg/actor/deploy_test.go:40]: 👍

File: pkg/actor/deploy.go
Comment (chrisseto) [inline pkg/actor/deploy.go:166]: nit: Inline error check?

File: pkg/actor/deploy.go
Comment (chrisseto) [inline pkg/actor/deploy.go:56]: Status gets set to installing if it's already installing?

Will this happen in the case of an error when installing?

File: api/v1alpha1/cluster_types.go
Comment (chrisseto) [inline api/v1alpha1/cluster_types.go:27]: OOC: Are these values required by RH or are we free to customize them?

**Review by chrisseto — DISMISSED** [review]
LGTM, could clean up tasks before merging please 🙏

File: api/v1alpha1/cluster_types.go
Comment (chrislovecnm) [reply api/v1alpha1/cluster_types.go:27]: We request some value, but we can set them to whatever we want.

File: pkg/actor/deploy.go
Comment (chrislovecnm) [reply pkg/actor/deploy.go:56]: We probably should do an error state.  Will file an issue to get an error state created.

**Comment by chrislovecnm** [comment]
@chrisseto need another review from you

**Review by chrisseto — DISMISSED** [review]
LGTM

File: pkg/controller/cluster_controller_test.go
Comment (chrislovecnm) [inline pkg/controller/cluster_controller_test.go:94]: @chrisseto any ideas?

File: pkg/testutil/fake_client.go
Comment (chrislovecnm) [inline pkg/testutil/fake_client.go:1]: TODO remove this file

File: pkg/testutil/fake.go
Comment (chrislovecnm) [inline pkg/testutil/fake.go:201]: TODO back out these changes.

File: pkg/testutil/BUILD.bazel
Comment (chrislovecnm) [inline pkg/testutil/BUILD.bazel:10]: TODO back out

File: pkg/controller/cluster_controller_test.go
Comment (chrisseto) [reply pkg/controller/cluster_controller_test.go:94]: Looks like it's just a noop test? Do you have the error handy by any chance?

**Comment by chrislovecnm** [comment]
Closing.  We have newer code

**Pull request closed without merging**
